libtiledbsoma/src/soma/managed_query.h

```diff
@@ -1028,6 +1027,28 @@ class ManagedQuery {
 
     template <typename T>
     std::vector<T> _enumeration_values_view(Enumeration& enumeration);
+
+    /**
+     * @brief Finds an enumeration value in a vector of already
+     * existing enumeration values. This helper method support
+     * comparing NaN values. Since NaN != NaN, we cannot use
+     * std::find. Also since TileDB and Arrow treat NaNs with
+     * different bit patterns as distinct values, we also cannot
+     * rely onstd::isnan and must use to bitwise comparisons.
+     *
+     * @param values the vector of values to search through
+     * @param target the value to search for
+     * @return typename std::vector<T>::const_iterator iterator
+     * to the found element or values.end() if not found
+     */
+    template <typename T>
+    typename std::vector<T>::const_iterator _find_enum_match(
```

Comment:
A thought about this -- it is O(N^2) in the worst case.  It will work great for most enum value lists, which are of modest size.

However, we occasionally get large (O(100K)) or even more rarely very large O(1M+) enum value lists.  The O(100K) lists would be ontologies like tissue (UBERON) or cell type (CL).

At some point we may need to consider more efficient algorithms, and potentially pick between them.

@XanthosXanthopoulos - will the upcoming Core changes exposing the underlying enum map address this?  I  believe it is covering this case, but flagging in case it doesn't